### ISSUE TITLE
maps saved with save/all can't be loaded with load/all

### ISSUE THREAD

**Issue by IngoHohmann** [metaeducation/ren-c#538]
```
>> save/all %test.r make map! ["a" 1] 

>> load/all %test.r                   
** Script Error: key must be LOCK-ed to add to MAP! "a"
** Where: transcode case load
** Near: data :file :line ?? take/last data
** File: ../src/core/t-map.c
** Line: 322
```
Version:
```
**    Version:   2.102.0.4.2                                            **
**    Platform:  Linux libc6-2-3-x86                                    **
**    Build:     8-Jun-2017/14:43:45                                    **
**    Commit:    126ed58d1975b375695cd3bc35ac1399ca12b17a               **
```

**Comment by hostilefork** [comment]
Hmmm, interesting.  :-/

Well, this looks like another one of those issues of a piece of information on a value that affects it having trouble in round-tripping if it's not somehow slipped into the data.  We have a few examples of that:

    >> save/all %test.reb next [a b c]

    >> print read/string %test.reb
    #[block! [[a b c] 2]]

    >> back load %test.reb
    == [a b c]

So the position has to be put in a way that "looks weird".  The locked or unlocked state of something hasn't "mattered" enough before to think of how to work it into such a syntax.

A lot gets lost, like identity of things...

     foo: copy "shared"
     m: make map! compose [1 (foo) 2 (foo)]

Before you save it, the identity of the shared string is such that appending to m/1 will also change m/2.  But when you reload it that aspect is lost.  So I've been increasingly skeptical of generic text-based serialization (e.g. where you don't have to be somewhat involved in writing serialization methods).  You lose binding information and such also.

And I imagine the "simple" answer (have /ALL record the locked status of any series in a construction syntax location) might wind up giving you something uglier than you wanted to see.

**Comment by IngoHohmann** [comment]
Well, for me, most of the time saving und being able to load is more important.
From time to time I might like to be able to look at the data in a text-editor.

**Comment by codebybrett** [comment]
Probably for the reasons mentioned above, I cannot remember if I've ever used construction syntax. I tend to view serialised files as unbound blocks (i.e. just tokens) that need a function to interpret. So I output them as needed and I like to be able to edit them easily, play with search and replace upon them and have their format clearly show their structure. Especially for ad-hoc work, I can bind them to whatever context is useful and run them in Do or interpret them otherwise.

I don't doubt that it would be useful to try to serialise some in-memory states, but I wonder if construction syntax is the right way. Instead of specialised construction syntax, perhaps we want an efficient dialect whose effect is to create particular in-memory structures - if such a thing is possible. If in future we want to attach user defined attributes to values (maybe I mean "tokens" or "symbols"), then it may be a matter of adding a new operation to the dialect instead of trying to invent a new syntax.

**Comment by hostilefork** [comment]
@codebybrett Yes, I've begun to agree that construction syntax may be the wrong way to think of things.  We might want to go down the route of helping people do plain old serialization methods, or think about how to dump out binary representations of the memory state that can round trip but be more like a Smalltalk VM state or such.
